Index: app/src/main/java/com/cs/meetingbridge/DiscussionFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/cs/meetingbridge/DiscussionFragment.java	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/java/com/cs/meetingbridge/DiscussionFragment.java	(revision )
@@ -37,15 +37,17 @@
         return dis;
     }
 
-
     @Override
     public View onCreateView(final LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        View rootView = inflater.inflate(R.layout.fragment_discussion, container, false);
+        final View rootView = inflater.inflate(R.layout.fragment_discussion, container, false);
         final ListView postListView = (ListView) rootView.findViewById(R.id.postsListView);
+
+
         Bundle bundle = this.getArguments();
         if (bundle != null) {
+
             final String id = bundle.getString("id");
             Toast.makeText(getActivity(), id, Toast.LENGTH_SHORT).show();
             databaseReference.child("Groups").addValueEventListener(new ValueEventListener() {
@@ -80,15 +82,13 @@
             createPost.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-
                     Intent intent = new Intent(getActivity(), CreatePostActivity.class);
                     intent.putExtra("id", id);
                     startActivity(intent);
+
                 }
             });
         }
-
-
         return rootView;
     }
 
Index: app/src/main/res/layout/activity_create_post.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_create_post.xml	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/res/layout/activity_create_post.xml	(revision )
@@ -31,4 +31,4 @@
         android:layout_margin="@dimen/fab_margin"
         app:srcCompat="@android:drawable/ic_dialog_email" />
 
-</android.support.design.widget.CoordinatorLayout>
+</android.support.design.widget.CoordinatorLayout>
\ No newline at end of file
Index: app/src/main/java/com/cs/meetingbridge/TimePickerFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/cs/meetingbridge/TimePickerFragment.java	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/java/com/cs/meetingbridge/TimePickerFragment.java	(revision )
@@ -26,7 +26,9 @@
     public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
         //Do something with the user chosen time
         //Get reference of host activity (XML Layout File) TextView widget
+
         TextView tv = (TextView) getActivity().findViewById(R.id.postTimeTV);
+        tv.setText("String");
         String min = String.valueOf(minute);
         //Set a message for user
         if (minute < 10) {
@@ -54,6 +56,7 @@
         }
         //Display the user changed time on TextView
         tv.setText(currentH + " " + min + " " + aMpM);
+        System.out.println(currentH + " " + min + " " + aMpM);
     }
 
 }
Index: app/src/main/java/com/cs/meetingbridge/DatePickerFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/cs/meetingbridge/DatePickerFragment.java	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/java/com/cs/meetingbridge/DatePickerFragment.java	(revision )
@@ -30,6 +30,7 @@
 
     public void onDateSet(DatePicker view, int year, int month, int day) {
         //Do something with the date chosen by the user
+
         TextView tv = (TextView) getActivity().findViewById(R.id.postDateTV);
         String mon = "";
         String d = String.valueOf(day);
\ No newline at end of file
Index: app/src/main/java/com/cs/meetingbridge/PostInfo.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/cs/meetingbridge/PostInfo.java	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/java/com/cs/meetingbridge/PostInfo.java	(revision )
@@ -5,11 +5,14 @@
     private String postId, postTitle, postDescription, postingTime;
     private PostTime postTime;
     private PostDate postDate;
+    private GroupInfo groupInfo;
 
+
     public PostInfo() {
     }
 
-    public PostInfo(String postId, String postTitle, String postDescription, PostTime postTime, PostDate postDate, userInfo host, String postingTime) {
+    public PostInfo(String postId, String postTitle, String postDescription,
+                    PostTime postTime, PostDate postDate, userInfo host, String postingTime, GroupInfo groupInfo) {
         this.postTitle = postTitle;
         this.postId = postId;
         this.postDescription = postDescription;
@@ -17,6 +20,15 @@
         this.postDate = postDate;
         this.host = host;
         this.postingTime = postingTime;
+        this.groupInfo = groupInfo;
+    }
+
+    public GroupInfo getGroupInfo() {
+        return groupInfo;
+    }
+
+    public void setGroupInfo(GroupInfo groupInfo) {
+        this.groupInfo = groupInfo;
     }
 
     public String getPostingTime() {
Index: app/src/main/java/com/cs/meetingbridge/PostListAdapter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/cs/meetingbridge/PostListAdapter.java	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/java/com/cs/meetingbridge/PostListAdapter.java	(revision )
@@ -1,13 +1,28 @@
 package com.cs.meetingbridge;
 
+import android.app.Dialog;
 import android.content.Context;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.BaseAdapter;
+import android.widget.Button;
+import android.widget.EditText;
 import android.widget.ImageView;
+import android.widget.ListView;
 import android.widget.TextView;
+import android.widget.Toast;
 
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.text.SimpleDateFormat;
 import java.util.ArrayList;
+import java.util.Date;
 
 public class PostListAdapter extends BaseAdapter {
     private Context mContext;
@@ -34,10 +49,10 @@
     }
 
     @Override
-    public View getView(int i, View view, ViewGroup viewGroup) {
-        View v = View.inflate(mContext, R.layout.post_layout, null);
+    public View getView(final int i, View view, ViewGroup viewGroup) {
+        final View v = View.inflate(mContext, R.layout.post_layout, null);
         PostTime time = mPostList.get(i).getPostTime();
-        PostDate date = mPostList.get(i).getPostDate();
+        final PostDate date = mPostList.get(i).getPostDate();
         TextView postingTime = (TextView) v.findViewById(R.id.postingTime);
 
         String timeString = time.getHours() + ":" + time.getMinutes() + " " + time.getAmpm();
@@ -56,6 +71,83 @@
         postingTime.setText(mPostList.get(i).getPostingTime());
         v.setTag(mPostList.get(i).getPostId());
 
+        Button commentButton = (Button) v.findViewById(R.id.commentsButton);
+
+
+        commentButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Toast.makeText(v.getContext(), mPostList.get(i).getGroupInfo().getGroupId(), Toast.LENGTH_SHORT).show();
+                final Dialog dialog = new Dialog(v.getContext());
+                dialog.setContentView(R.layout.comment_dialog_layout);
+                dialog.setTitle(mPostList.get(i).getGroupInfo().getGroupName());
+                final EditText commentBox = (EditText) dialog.findViewById(R.id.postCommentET);
+                final ListView commentLV = (ListView) dialog.findViewById(R.id.commentsLV);
+                Button dismissButton = (Button) dialog.findViewById(R.id.dismiss);
+                final Button postCommentButton = (Button) dialog.findViewById(R.id.postCommentButton);
+
+                final DatabaseReference databaseReference = FirebaseDatabase.getInstance().getReference();
+                dismissButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        dialog.dismiss();
+                    }
+                });
+                final FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
+                final String currentTime = new SimpleDateFormat("dd-M-yy hh:mm a").format(new Date());
+                databaseReference.child("users").child(currentUser.getUid()).addValueEventListener(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+                        final userInfo user = dataSnapshot.getValue(userInfo.class);
+                        databaseReference.child("Comments")
+                                .child(mPostList.get(i).getGroupInfo().getGroupId())
+                                .child(mPostList.get(i).getPostId())
+                                .addValueEventListener(new ValueEventListener() {
+                                    @Override
+                                    public void onDataChange(DataSnapshot dataSnapshot) {
+                                        final ArrayList<CommentInfo> commentInfos = showComments(dataSnapshot);
+                                        final CommentListAdapter adapter = new CommentListAdapter(v.getContext(), commentInfos);
+                                        commentLV.setAdapter(adapter);
+                                        postCommentButton.setOnClickListener(new View.OnClickListener() {
+                                            @Override
+                                            public void onClick(View v) {
+                                                commentInfos.add(0, new CommentInfo(currentTime, user, commentBox.getText().toString(), mPostList.get(i).getGroupInfo()));
+                                                databaseReference.child("Comments")
+                                                        .child(mPostList.get(i).getGroupInfo().getGroupId())
+                                                        .child(mPostList.get(i).getPostId()).setValue(commentInfos);
+                                                commentBox.setText("");
+                                                commentLV.setAdapter(adapter);
+
+                                            }
+                                        });
+                                    }
+
+                                    @Override
+                                    public void onCancelled(DatabaseError databaseError) {
+
+                                    }
+                                });
+                    }
+
+                    @Override
+                    public void onCancelled(DatabaseError databaseError) {
+
+                    }
+                });
+                dialog.show();
+
+            }
+        });
+
         return v;
+    }
+
+    private ArrayList<CommentInfo> showComments(DataSnapshot dataSnapshot) {
+        ArrayList<CommentInfo> commentInfo = new ArrayList<>();
+        for (DataSnapshot data : dataSnapshot.getChildren()) {
+            CommentInfo c = data.getValue(CommentInfo.class);
+            commentInfo.add(c);
+        }
+        return commentInfo;
     }
 }
Index: app/src/main/res/layout/content_create_post.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/content_create_post.xml	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/res/layout/content_create_post.xml	(revision )
@@ -11,11 +11,11 @@
     android:paddingRight="@dimen/activity_horizontal_margin"
     android:paddingTop="@dimen/activity_vertical_margin"
     app:layout_behavior="@string/appbar_scrolling_view_behavior"
-    tools:context="com.cs.meetingbridge.CreatePostActivity"
+    tools:context=".CreatePostActivity"
     tools:showIn="@layout/activity_create_post">
 
     <EditText
-        android:id="@+id/postTitle"
+        android:id="@+id/postTitleET"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:hint="Title"
@@ -30,7 +30,8 @@
     <TextView
         android:id="@+id/postTimeTV"
         android:layout_width="wrap_content"
-        android:layout_height="wrap_content" />
+        android:layout_height="wrap_content"
+        android:text="Event Time A" />
 
     <Button
         android:id="@+id/setDate"
@@ -41,10 +42,11 @@
     <TextView
         android:id="@+id/postDateTV"
         android:layout_width="wrap_content"
-        android:layout_height="wrap_content" />
+        android:layout_height="wrap_content"
+        android:text="Event Date a" />
 
     <EditText
-        android:id="@+id/postDescriptionTV"
+        android:id="@+id/postDescriptionET"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:hint="Post Details"
\ No newline at end of file
Index: app/src/main/res/layout/fragment_discussion.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/fragment_discussion.xml	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/res/layout/fragment_discussion.xml	(revision )
@@ -13,6 +13,7 @@
         android:layout_height="wrap_content"
         android:text="New Post" />
 
+
     <ListView
         android:id="@+id/postsListView"
         style="@android:style/Widget.DeviceDefault.ExpandableListView"
Index: app/src/main/res/layout/post_layout.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/post_layout.xml	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/res/layout/post_layout.xml	(revision )
@@ -55,4 +55,10 @@
         android:layout_height="wrap_content"
         android:text="Date"
         android:textSize="15dp" />
+
+    <Button
+        android:id="@+id/commentsButton"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Comments" />
 </LinearLayout>
\ No newline at end of file
Index: app/src/main/res/layout/comment_dialog_layout.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/comment_dialog_layout.xml	(revision )
+++ app/src/main/res/layout/comment_dialog_layout.xml	(revision )
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="350dp"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+
+    <Button
+        android:id="@+id/dismiss"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="end"
+        android:background="@null"
+        android:text="X" />
+
+    <EditText
+        android:id="@+id/postCommentET"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_gravity="start"
+        android:hint="Write a Comment!" />
+
+    <Button
+        android:id="@+id/postCommentButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="end"
+        android:text="Post" />
+
+
+    <ListView
+        android:id="@+id/commentsLV"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content" />
+
+
+</LinearLayout>
\ No newline at end of file
Index: app/src/main/java/com/cs/meetingbridge/CreatePostActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/cs/meetingbridge/CreatePostActivity.java	(revision 7d80266c85bca03276f2bd1f1bb78c9a90b90c24)
+++ app/src/main/java/com/cs/meetingbridge/CreatePostActivity.java	(revision )
@@ -1,3 +1,4 @@
+
 package com.cs.meetingbridge;
 
 import android.content.Intent;
@@ -38,8 +39,8 @@
         setSupportActionBar(toolbar);
 
         databaseReference = FirebaseDatabase.getInstance().getReference();
-        final EditText postTitle = (EditText) findViewById(R.id.postTitle);
-        final EditText postDescription = (EditText) findViewById(R.id.postDescriptionTV);
+        final EditText postTitle = (EditText) findViewById(R.id.postTitleET);
+        final EditText postDescription = (EditText) findViewById(R.id.postDescriptionET);
         Button setTimeButton = (Button) findViewById(R.id.setTime);
         Button setDateButton = (Button) findViewById(R.id.setDate);
         final TextView timeView = (TextView) findViewById(R.id.postTimeTV);
@@ -97,14 +98,15 @@
                 databaseReference.child("users").child(currentUser.getUid()).addValueEventListener(new ValueEventListener() {
                     @Override
                     public void onDataChange(DataSnapshot dataSnapshot) {
-                        userInfo user = dataSnapshot.getValue(userInfo.class);
-                        final PostInfo postInfo = new PostInfo("1", title, description, postTime, postDate, user, currentTime);
+                        final userInfo user = dataSnapshot.getValue(userInfo.class);
+
                         databaseReference.child("Groups").addValueEventListener(new ValueEventListener() {
                             @Override
                             public void onDataChange(DataSnapshot dataSnapshot) {
                                 ArrayList<GroupInfo> groupInfos = getCurrentGroup(dataSnapshot);
                                 int temp = Integer.parseInt(id);
                                 final String groupId = groupInfos.get(temp).getGroupId();
+                                final PostInfo postInfo = new PostInfo("1", title, description, postTime, postDate, user, currentTime, groupInfos.get(temp));
                                 databaseReference.child("Posts").child(groupId).push().setValue(postInfo).addOnSuccessListener(new OnSuccessListener<Void>() {
                                     @Override
                                     public void onSuccess(Void aVoid) {
@@ -158,6 +160,7 @@
                 databaseReference.child("Posts").child(groupId).child(k).setValue(p);
                 break;
             }
+
         }
     }
 
@@ -176,4 +179,4 @@
     }
 
 
-}
+}
\ No newline at end of file
